gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+### Minimum Viable Product
+
+While this proposal defines a smart contracts based solution using Graph Horizon, we also propose an MVP that can be built without any smart contracts changes using off-chain interfaces between Indexers and Gateways. This will allow us to implement parts of the solution that will be useful for the full proposal implementation, will also allow us to learn from seeing the mechanism in operation, and will help define the user interfaces and pricing approaches for subgraph developers that can be built on top of the protocol. As the MVP does not require changes to the on-chain protocol, it technically does not require a formal GIP approval, but we present it here alongside the full proposal to gather consensus towards this approach and to invite Indexers to participate in this initial stage and provide feedback.
+
+In this MVP, we will roll out a new gateway component (the “dipper”) that will be controlled with a CLI. Gateway operators can be onboarded to this CLI and use it to add subgraphs to the list of supported subgraphs. Once a subgraph is in the list, the dipper will take care of finding Indexers and setting up indexing agreements with them.
+
+The Indexer stack will be updated to be able to accept indexing agreements from the dipper when the price is above the price set by the Indexer, and up to a maximum amount of subgraphs also set by the Indexer.
+
+Pricing between Indexer and gateway in the MVP will be based on the "amount of work done" proxy detailed above.
+
+The dipper will initially set up agreements with 3 Indexers for each desired subgraph, as we estimate this is a reasonable number to achieve 99.9% uptime in most subgraphs.
+
+Once the smart contracts proposed in this GIP are deployed, we anticipate adapting the indexer and gateway components accordingly.
+
+The MVP has different trust implications than what this proposal has described so far. In particular these are:
+
+- Indexers do the work upfront and trust that the gateway will pay them accordingly after the fact, since the gateway issues payment vouchers upon receiving PoIs and not before.
+- Gateways trust that indexers are reporting the correct amount of work done, since this information is not posted on-chain and thus it's non-slashable. This is partially mitigated by the fact that the gateway can (theoretically) compare the reported work done among the pool of indexers selected and thus can weed out bad actors, but it's nevertheless a weaker mechanism than slashing.
+
+## Detailed Specification
+
+TODO
+
+## Dependencies
+
+This GIP depends on Horizon being fully functional.
+
+## Rationale and Alternatives
+
+When presenting this proposal, we often hear suggestions for alternative, and possibly simpler, approaches to payments, for instance:
```

Comment:
```suggestion
Alternative (possibly simpler) approaches have been considered, including:
```

Reply:
@pcarranzav

Reply:
I prefer the original, we may reword it a bit but I wanted to explain that these are alternative proposals that we were asked about rather than things we considered ourselves.